packages/common/src/i18n/format_date.ts

```diff
@@ -382,8 +382,16 @@ function weekGetter(size: number, monthBased = false): DateFormatter {
       const today = date.getDate();
       result = 1 + Math.floor((today + nbDaysBefore1stDayOfMonth) / 7);
     } else {
-      const firstThurs = getFirstThursdayOfYear(date.getFullYear());
+      let firstThurs = getFirstThursdayOfYear(date.getFullYear());
       const thisThurs = getThursdayThisWeek(date);
+      /*
+       If some days of a year are part of next year according
+       to ISO 8601. Change the firstThurs for those days to next
+       year firstThurs
+      */
+      if (thisThurs.getFullYear() !== firstThurs.getFullYear()) {
+        firstThurs = getFirstThursdayOfYear(date.getFullYear() + 1);
+      }
```

Comment:
I believe we can make this logic much simpler...

```suggestion
      const thisThurs = getThursdayThisWeek(date);
      // Some days of a year are part of next year according to ISO 8601.
      // Compute the firstThurs from the year of this week's Thursday
      const firstThurs = getFirstThursdayOfYear(thisThurs.getFullYear());
```

WDYT? Am I missing something?

Reply:
seems much better to me